src/lib/pick-defined-values.ts

```diff
@@ -0,0 +1,42 @@
+import { isString } from '@guardian/libs';
+
+type ValidTargeting<T, K> = '' extends T
+	? never
+	: [] extends T
+	? never
+	: [''] extends T
+	? never
+	: T extends boolean
+	? never
+	: T extends NonNullable<T>
+	? K
+	: never;
+type DefinedKeys<T> = { [K in keyof T]-?: ValidTargeting<T[K], K> }[keyof T];
+type ObjectWithDefinedValues<T> = Pick<T, DefinedKeys<T>>;
+
+const isValidTargeting = (value: unknown): value is string | string[] => {
+	if (isString(value) && value !== '') return true;
+	if (
+		Array.isArray(value) &&
+		value.filter(isString).filter(Boolean).length > 0
+	)
+		return true;
```

Comment:
'Dangling' returns scare me a bit.

Also can we improve the readability a bit and create named variables for the boolean conditions?